symfony/plugins/orangehrmCoreOAuthPlugin/Api/OAuthClientAPI.php

```diff
@@ -0,0 +1,222 @@
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\OAuth\Api;
+
+
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\CrudEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\Exception\RecordNotFoundException;
+use OrangeHRM\Core\Api\V2\Model\ArrayModel;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Exception\DaoException;
+use OrangeHRM\Entity\OAuthClient;
+use OrangeHRM\OAuth\Api\Model\OAuthClientModel;
+use OrangeHRM\OAuth\Constant\GrantType;
+use OrangeHRM\OAuth\Constant\Scope;
+use OrangeHRM\OAuth\Dto\OAuthClientSearchFilterParams;
+use OrangeHRM\OAuth\Service\OAuthService;
+use OrangeHRM\Core\Api\V2\EndpointCollectionResult;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+
+
+class OAuthClientAPI extends Endpoint implements CrudEndpoint
+{
+
+    public const PARAMETER_CLIENT_ID = 'clientId';
+    public const PARAMETER_CLIENT_SECRET = 'clientSecret';
+    public const PARAMETER_REDIRECT_URI = 'redirectUri';
+
+    /**
+     * @var null|OAuthService
+     */
+    protected ?OAuthService $oAuthService = null;
+
+
+    /**
+     * @return OAuthService
+     */
+    public function getOAuthService(): OAuthService
+    {
+        if (is_null($this->oAuthService)) {
+            $this->oAuthService = new OAuthService();
+        }
+        return $this->oAuthService;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        $oAuthClientSearchFilterParams = new OAuthClientSearchFilterParams();
+        $this->setSortingAndPaginationParams($oAuthClientSearchFilterParams);
+
+        $count = $this->getOAuthService()->getOAuthClientDao()->getOAuthClientsCount($oAuthClientSearchFilterParams);
+
+        $oauthClients = $this->getOAuthService()->getOAuthClientDao()->getOAuthClients($oAuthClientSearchFilterParams);
+        return new EndpointCollectionResult(
+            OAuthClientModel::class,
+            $oauthClients,
+            new ParameterBag([CommonParams::PARAMETER_TOTAL => $count])
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            ...$this->getSortingAndPaginationParamsRules(OAuthClientSearchFilterParams::ALLOWED_SORT_FIELDS)
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function create(): EndpointResult
+    {
+        // TODO:: Check data group permission
```

Comment:
if you are keen to add data-group permission .. it's supporting now .. 
anyway .. nice-to-have thing :)

Reply:
remove this comment

Reply:
done